### PR TITLE
DEVENGAGE-801

### PR DESCRIPTION
**Pull request by themooks**
Content for DEVENGAGE-801

### CODE REVIEW

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+Make sure to ***Activate*** this add-on (it will then appear as ***Active***).
+
+![BYOC Cloud Add-On](AddOnBYOC01.png)
+
+
+### Step 2 - Prepare and collect information about your external SIP device 
+
+This information will be required on Genesys Cloud side, when configuring the new BYOC Cloud trunk.
+
+1. __SIP Server/Proxy address__. Address [*IP Address(es) or FQDN(s) and IP Address(es)*], Port [*SIP port*] and Transport Protocol [*UDP/TCP/TLS*] of your SIP Server/Proxy.
```
Comment (timsmithgenesys) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:53]: Don't use square brackets unless you're making a markdown link. Square brackets by themselves are still parsed as shown as links, but they won't have a valid href attribute as they're missing the link address.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+  We need at least one FQDN (Fully Qualified Domain Name) or a public IP address to target your SIP Server/Proxy when transferring calls from Genesys Cloud.
+
+  An IP Address is also required to define the SIP Access Control List on Genesys BYOC Cloud trunk side (IP or CIDR Address(es) of your SIP Server/Proxy). The SIP Access Control List allows to control and to limit access to the SIP service.
+
+  *Ex: 10.34.12.16 on port 5060, using UDP as transport protocol for SIP*
```
Comment (timsmithgenesys) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:59]: I'll enter this comment into the running for _Most Pretentious Review Comment_:

This should be `E.g.` instead of `Ex`. https://blog.prepscholar.com/ie-vs-eg-vs-ex-definition 

TLDR;
* `i.e.` - short for the latin phrase "id est", meaning "in other words" and the clarification is a full replacement for the subject. 
* `e.g.` - short for the latin phrase "exempli gratia", meaning "for example" and the clarification is a subset of valid examples.
* `e.x.` - short for the english word "exercise", used to reference a specific task elsewhere in the document

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+  Note that if you plan to use TLS as Transport Protocol for SIP, it implies that your SIP Server/Proxy must leverage a FQDN (Fully Qualified Domain Name), and that you have a valid certificate, signed by a recognized Certification Authority (CA). See [__TLS trunk transport protocol specification__](https://help.mypurecloud.com/?p=197875).
+
+2. __Supported Audio codecs__. Audio codecs that your external device supports and that you want to enforce.
+
+  i.e. G.711 ulaw/alaw (PCMU/PCMA) only? Opus as well?
```
Comment (timsmithgenesys) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:65]: In the running for second most pretentious comment, this should probably be `e.g.` as well, unless that is the only supported codec. Also looks like this statement was intended to be rewritten once an answer was found.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 1 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: Thanks. I had missed this one - wanted to use parenthesis and don't know why, it ended with square brackets.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 2 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: Ex: is a french abbreviation for "exemple". Changed them to e.g.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 3 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: Changed to e.g.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+At this time, only Genesys Cloud Voice and BYOC Cloud can coexist in a single Genesys Cloud org.
+The enterprise could then have a Genesys Cloud Voice Trunk (for PSTN connectivity) and a BYOC Cloud Trunk (to interact with your external SIP device).  
+Or the enterprise could have a BYOC Cloud Trunk (for PSTN connectivity via a PSTN/SIP provider) and a second BYOC Cloud Trunk (to interact with your external SIP device).
+
+:::primary
+**Note:** Combining Genesys Cloud Voice or BYOC Cloud with a BYOC Premise Trunk is not supported at this stage.
+Later this year, Genesys Cloud will support Hybrid environments, allowing to combine Genesys Cloud Voice, BYOC Cloud and BYOC Premise.
+:::
+
+About:
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:27]: This suddenly came out out of nowhere.  You might want to add a sentence explaining why this reference material is useful for the immediate discussion.  If it's just additional resources, I would add an additional resource section at the very bottom of the post and put it there.  I also like to put a 1 sentence description next to each resource so the reader understands what they are going to get out of clicking on the link.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+- [__Genesys Cloud Voice__](https://help.mypurecloud.com/?p=60252)
+- [__BYOC Cloud__](https://help.mypurecloud.com/?p=152702)
+
+The following article assumes that you have setup your connectivity with the PSTN, using Genesys Cloud Voice or BYOC Cloud: *Location(s) with verified address(es); Site(s) using Cloud Media Model, one set as Default Site, and with Outbound Routes that leverage the Genesys Cloud Voice pre-defined trunk or the BYOC Cloud trunk used for PSTN connectivity*.
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:31]: You might want to call this out using an alert box and potentially move it to the beginning of the post.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 2 above)
Comment (carnellj) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: I did not know that (Ex.). Good to know :)

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+
+![Trunk step 11](Trunk11.png)
+
+On outbound calls (Genesys Cloud to External via BYOC Cloud), Genesys Cloud will add a ***x-inin-cnv*** header containing the ConversationId on SIP INVITE.  
+For inbound calls (External to Genesys Cloud via BYOC Cloud), Genesys Cloud will add a ***x-inin-cnv*** header containing the ConversationId in the 200 OK (following the received SIP INVITE).
+
+
+### Data as User to User Information 
+
+If you need to pass any additional data from Genesys Cloud to your external device, you can also leverage User to User Information (UUI).
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:257]: Should we build User to User Information?  It would seem to indicate it is a noun that is being called out.  Not sure.,

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+It is the protocolCallId attribute, in the session structure (at participant level).
+Note: There are multiple CallIDs in a call (one for each leg/session).
+
+You can get the SIP Call-ID from a ConversationId using a [GET /api/v2/analytics/conversations/{conversationId}/details](/api/rest/v2/analytics/#get-api-v2-analytics-conversations--conversationId--details).  
+Depending on your use case and flow, find the "inbound" or "outbound" participant (*purpose* = customer or external) and check the *protocolCallId* in his session.
+
+You can also find the ConversationId from a SIP Call-ID using a [query for conversation details - POST /api/v2/analytics/conversations/details/query](/api/rest/v2/analytics/#post-api-v2-analytics-conversations-details-query).  
+In your query, define a *segmentFilter* that leverages the *protocolCallId* dimension.
+
+Also note that the Analytics Conversation Details are available right after the conversation starts. You do not need to wait for the end of a conversation to perform an Analtyics query for conversation details.
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: Should Analytics Conversation Details be bolded

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+
+
+## Controlling access to your SIP Device 
+
+In order to control and to limit the access to your external device, which will be exposed on the public internet, you can leverage a combination of the following methods. These are up to you - nothing is mandatory!
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:321]: You should probably list out the methods even if you are going to go into detail.  Otherwise, it feels like you have dropped the subject even though you are about to talk about them.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+
+In the trunk configuration, under ***Protocol***, in ***Outbound***, you can define the custom SIP header and the value (static) that will be added to the SIP INVITE, as described in the previous sections.
+
+
+### Token in UUI data (User to User Information) 
+
+It is also possible to leverage the UUI data, which can be set at the Architect Inbound Call flow level, to carry a token, assigned to your Genesys Cloud customer, as an additional control and verification mechanism.
+
+See [Data as User to User Information](#data-as-user-to-user-information), under [Sending data on transfer to your SIP device](#sending-data-on-transfer-to-your-sip-device), for more details.
+
```
Comment (carnellj) [inline source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:351]: I would add another section on Closing Thoughts.   Maybe bring into this section the most common mistakes people make in setting up trunks or your experiences.  Without closing thoughts to tie the article together you are giving a lot of good information, but not wrapping it up.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+date: 2021-09-03
+author: jerome.saint-marc
+category: 3
+image: BYOCCloudtoExternalDevice.png
+---
+
+The purpose of this article is to describe how to create and to configure a [BYOC Cloud Trunk](https://help.mypurecloud.com/?p=152702) (SIP Trunk):
+- to transfer and to connect calls from Genesys Cloud to an external system (that supports SIP connectivity over the public internet),
+- and to transfer these calls back to Genesys Cloud (releasing resources on the external SIP device).
+
```
[the remark this answers is not in the document]
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:13]: Changed.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+Typically, this trunk would coexist with another trunk (i.e. Genesys Cloud Voice or BYOC Cloud Trunk), used by the Genesys Cloud org (i.e. the enterprise), to manage inbound and outbound traffic with end customers via the PSTN.
```
[the remark this answers is not in the document]
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:14]: Changed.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+
+![Simplified view of Genesys Cloud and Trunks](BYOCCloudtoExternalDevice.png)
+
+At this time, only Genesys Cloud Voice and BYOC Cloud can coexist in a single Genesys Cloud org.
+The enterprise could then have a Genesys Cloud Voice Trunk (for PSTN connectivity) and a BYOC Cloud Trunk (to interact with your external SIP device).  
+Or the enterprise could have a BYOC Cloud Trunk (for PSTN connectivity via a PSTN/SIP provider) and a second BYOC Cloud Trunk (to interact with your external SIP device).
+
+:::primary
+**Note:** Combining Genesys Cloud Voice or BYOC Cloud with a BYOC Premise Trunk is not supported at this stage.
+Later this year, Genesys Cloud will support Hybrid environments, allowing to combine Genesys Cloud Voice, BYOC Cloud and BYOC Premise.
```
[the remark this answers is not in the document]
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:24]: I don't know exactly. The beta was recently announced on the Genesys Cloud Community forum: https://community.genesys.com/digestviewer29/viewthread?GroupId=19&MessageKey=ef6c18f7-f783-4b3b-b0d6-0c80ca7e20f2&CommunityKey=bab95e9c-6bbe-4a13-8ade-8ec0faf733d4&tab=digestviewer

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 4 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:27]: Removed. I have added the links in the text which was referencing Genesys Cloud Voice and BYOC Trunk in previous section.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 5 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:31]: Done.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 6 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:257]: I have added the link to the Resource Center on User to User Information (UUI).

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 7 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md]: Changed.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 8 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:321]: Added.

File: source/2021-09-03-transferring-calls-to-external-using-byoc/index.md (hunk 9 above)
Comment (themooks) [reply source/2021-09-03-transferring-calls-to-external-using-byoc/index.md:351]: Done - or at least I tried :-)

**Review by carnellj — APPROVED** [review]
